pyoptsparse/pyOpt_utils.py

```diff
@@ -529,3 +529,13 @@ def _csc_to_coo(mat: dict) -> dict:
     coo_data = np.array(data)
 
     return {"coo": [coo_rows, coo_cols, coo_data], "shape": mat["shape"]}
+
+
+def _broadcast_to_array(name: str, value, n_vars: int, allow_none: bool = True):
+    try:
+        value = np.broadcast_to(value, (n_vars))
```

Comment:
Arrays containing None will get broadcast into `dtype=object`. I think this is fine since we only examine individual components of these arrays, e.g. [here](https://github.com/mdolab/pyoptsparse/blob/main/pyoptsparse/pyOpt_constraint.py#L121) for constraints. Design variables are always stored in scalar form so it doesn't matter there.

Reply:
I believe there is no change in behaviour, since we iterate over the object so it doesn't matter if it's an array or list containing `None`.